Sources/PackageCollections/PackageCollections.swift

```diff
@@ -294,6 +350,36 @@ public struct PackageCollections: PackageCollectionsProtocol {
             return PackageCollectionsModel.TargetListItem(target: pair.target, packages: targetPackages)
         }
     }
+
+    // TODO: confirm this is the correct logic 👀
+    internal func mergedPackageMetadata(package: PackageCollectionsModel.Collection.Package,
+                                        basicMetadata: PackageCollectionsModel.PackageBasicMetadata?) -> PackageCollectionsModel.Package {
+        var versions = package.versions.map { packageVersion in
+            PackageCollectionsModel.Package.Version(version: packageVersion.version,
+                                                    packageName: packageVersion.packageName,
+                                                    targets: packageVersion.targets,
+                                                    products: packageVersion.products,
+                                                    toolsVersion: packageVersion.toolsVersion,
+                                                    verifiedPlatforms: packageVersion.verifiedPlatforms,
+                                                    verifiedSwiftVersions: packageVersion.verifiedSwiftVersions,
+                                                    cves: nil, // FIXME: where do we get this from?
+                                                    license: packageVersion.license)
+        }
+
+        // FIXME: this is naive sorting, need to implement per spec
```

Comment:
@yim-lee 👀

Reply:
The spec is this?

```
        /// The latest published version of the package
        ///
        /// - Note:
        ///     This would be the latest released version, unless no release versions are published
        ///     in which case it will be the latest pre-release version.
        ///
        ///     E.g. given:
        ///     3.0.0-beta.1
        ///     2.1.1
        ///     2.1.0
        ///     2.0.0
        ///     2.0.0-beta.2
        ///     2.0.0-beta.1
        ///     1.0.1
        ///     1.0.0
        ///
        ///     Latest =  2.1.1
        ///
        ///     And given:
        ///     1.0.0-beta.3
        ///     1.0.0-beta.2
        ///     1.0.0-beta.1
        ///
        ///     Latest = 1.0.0-beta.3
```

Reply:
yea

Reply:
@abertelrud @neonichu is there some code in SwiftPM/TSC that already does this logic?

Reply:
https://github.com/apple/swift-tools-support-core/blob/main/Sources/TSCUtility/Version.swift#L54